base/math.jl

```diff
@@ -538,6 +544,26 @@ Stacktrace:
 ```
 """
 hypot(x::Number, y::Number) = hypot(promote(x, y)...)
+
+hypot(x::Integer, y::Integer) = hypot(convert.(Float64, (x,y))...)
```

Comment:
```suggestion
hypot(x::Integer, y::Integer) = hypot(float(x), float(y))
```
will allow `BigInt`s to get converted to `BigFloat`s (which is the convention).

Reply:
You'll still need to promote them to the same float type in order to hit the optimized method below. See my suggestion (which we posted at the same time :O)